crates/kitsune_p2p/kitsune_p2p/tests/common/harness.rs

```diff
@@ -149,19 +150,45 @@ impl KitsuneTestHarness {
     }
 }
 
-pub async fn start_bootstrap() -> (SocketAddr, AbortHandle) {
+pub struct TestBootstrapHandle {
+    shutdown_cb: Option<BootstrapShutdown>,
+    abort_handle: AbortHandle,
+}
+
+impl TestBootstrapHandle {
+    fn new(shutdown_cb: BootstrapShutdown, abort_handle: AbortHandle) -> Self {
+        Self {
+            shutdown_cb: Some(shutdown_cb),
+            abort_handle,
+        }
+    }
+
+    pub fn abort(&mut self) {
+        if let Some(shutdown_cb) = self.shutdown_cb.take() {
+            shutdown_cb();
+        }
+        self.abort_handle.abort();
+    }
+}
+
+impl Drop for TestBootstrapHandle {
+    fn drop(&mut self) {
+        self.abort();
+    }
+}
+
+pub async fn start_bootstrap() -> (SocketAddr, TestBootstrapHandle) {
     let (bs_driver, bs_addr, shutdown) =
         kitsune_p2p_bootstrap::run("127.0.0.1:0".parse::<SocketAddr>().unwrap(), vec![])
             .await
             .expect("Could not start bootstrap server");
 
     let abort_handle = tokio::spawn(async move {
-        let _shutdown_cb = shutdown;
```

Comment:
This needs to be held but is meant to be called, not dropped. Not sure if this was causing a problem or not but getting a quick and clean shutdown matters to several tests so this is more correct